### DISCUSSION TITLE
特定のキー(ballside_sw24/Nキー)が反応しない

### DISCUSSION THREAD

**Discussion by Kenneth23ef** [Yowkees/keyball#113] [Q&A] [answered]
右手ballsideの基盤について、一度ダイオードの付け外しを実施した際にランドがいくつか取れたため、ジャンパー線を取り付けるなどしていくつかのキーが反応するようになりましたが、最終的にNキー(Row2/Col3、回路図の[SW24]の箇所)だけが反応しません。  

添付画像の青点の箇所同士/赤点の箇所同士はそれぞれテスターで導通が取れており、キースイッチを差し込みNキーを入力した状態で緑矢印の箇所も導通が取れております。またNキーのダイオードもテスターでは特に故障は確認できません(問題切り分けのため、新品のダイオードに取り替えたり、リードタイプのダイオードに取り替えたりもしましたが、状況変わらずです。画像は現在リードタイプのダイオードが取り付けられています)

このような状態で、他にどのような原因が考えられるでしょうか？  

ちなみに、青点のソケット端子と、promicro側の赤点(A/Row2のピン)を直接つないでみても、特に反応がありませんでした。  
同様に、青点のソケット端子と、promicro側のピンが刺さっていない箇所(A')も直接つないでみて、特に反応ありませんでした。  

参考にした回路図は以下になります。  
https://github.com/Yowkees/keyball/blob/main/keyball46/doc/rev1/Keyball46_Circuit_BallSide.pdf  

当方あまり電子回路に触れた機会が少なく、そもそもの確認方法が誤っていたら申し訳ありません...  
一番最初にダイオードを取り付けた際に、安物の中華製のはんだごてを使用しており、はんだづけの際に少し手こずった影響で基盤自体が破損してしまった可能性を少し考えています...(現在は温調付きのはんだごてを使用しております)

お忙しいところ申し訳ありませんが、ご確認いただけると助かります。

![IMG_20220418_010624](https://user-images.githubusercontent.com/33027558/163725097-9e663ed6-ea29-4060-9bca-505a1249f87f.jpg)

**Comment by Kenneth23ef** [comment]
4/23 追記

製作者の方にコンタクト取り、基盤を新たに購入することになりました。新基盤で試してみます。

**Reply by Yowkees** [reply]
完動に向けて頑張ってください！応援しています🙏

**Reply by Kenneth23ef** [reply]
4/27 追記

お世話になります。先日 #Keyball46 のballside基盤を制作者の方から新たに購入した者です。新しい基盤と、promicroとコンスルー12ピンをこちらで新たに購入して再度組み立てたのですが、状況変わらず、Nキーのみ反応しません。導通は前回同様問題なさそうです。他に考えられる原因はございますでしょうか？

以下のファームウェアを QMK ToolBox で書き込んでいます。  
[keyball_keyball46_via.hex](https://github.com/Yowkees/keyball/blob/main/qmk_firmware/.build/keyball_keyball46_via.hex)

テスト用ファームウェアで書き込んでみても、同様にNキーが反応しません...  
[keyball_keyball46_test.hex](https://github.com/Yowkees/keyball/blob/main/qmk_firmware/.build/keyball_keyball46_test.hex)

利用しているpromicro  
https://ja.aliexpress.com/i/2010847161.html

![IMG_20220427_192958](https://user-images.githubusercontent.com/33027558/165502200-c9e6e6a7-9338-40a7-a19c-c5294ca922d7.jpg)

度々の質問となり申し訳ありません。よろしくお願い致します。

**Reply by Yowkees** [reply]
了解しました。明日同じ状況を作って確認してみますので少々お待ちください🙏

**Reply by Kenneth23ef** [reply]
ありがとうございます。よろしくおねがいいたします🙇

**Reply by Yowkees** [reply]
質問があります！入力できないのは、Nキーだけでしょうか？

**Reply by Kenneth23ef** [reply, marked as answer]
5/1 追記

製作者の方に基盤をお送りし、調査頂いた結果、 
Keyball46はトラックボール基板を取り付けないとNキーが入力できない仕様とのことでした！
以下製作者の方のメッセージです。

> この様な動作となる理由は、Keyball46にはトラックボールあり基板となし基板が存在します。これを区別するためにUSBケーブルが接続された際にProMicroはトラックボールセンサーにアクセスし、反応が無ければ、自分はトラックボールなし基板だと判断します。
トラックボール無し基板が右手側だと仮定して見ていただくと、YキーとHキーが1.5UサイズでNキーに当たるキーが無いのです。そのため入力できませんでした。

**Reply by Yowkees** [reply]
追記ありがとうございます。
迷わないようにビルドガイドにも書きたいと思います！
